### PR TITLE
[WIP] Add layers in include to map by default

### LINKED ISSUE #349: Unearthing demo not visible in publiclab.org
Hi all, i'm not sure what happened, as we haven't published anything i think... but I don't see the markers on https://publiclab.org/wiki/unearthing-pvd any more. @sagarpreet-chadha @crisner @nstjean can we try to prioritize debugging this? 

Thank you -- i guess we need a system test in plots2 to ensure this stays live? There are a lot of moving parts!

### PR DESCRIPTION
**Pull request by crisner**
Fixes #349 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by crisner** [comment]
Work done:
- Added Unearthing layer to demo
- Fixed a bug where the points from the Unearthing layer weren't removed from the map after unchecking the layer on the layers control
- Fixed code to add all layers to map upon initialization if layers are passed through the include param
  - Now all layers would be visible on the map upon initialization
  - The status of the layers cannot be tracked for embed purposes. All the layers will always will be loaded on the map even if the embed code does not have the layers in its hash

#### Screenshot of unearthing-pvd.html upon initialization:
127 0 0 1_5500_example_unearthing-pvd html (1)

**Comment by crisner** [comment]
@jywarren, It looks like all layers are passed into the include param for the map page, publiclab.org/map here:
https://github.com/publiclab/plots2/blob/46af8030c82fe8346702edbaf917bfeb9e229163/app/views/map/map.html.erb#L104

We should either remove that line or add an extra option param to indicate whether we need the included layers to be displayed upon initialization.

**Comment by crisner** [comment]
Also, it looks like all layers are not included in the map page here: https://publiclab.org/map
Was then an intended change @jywarren, @sagarpreet-chadha, @nstjean ?

**Comment by nstjean** [comment]
I didn't do anything with the layers on that page, so I think it wasn't complete yet and they need to be added.

**Comment by jywarren** [comment]
Great work!!! I'll bump the version number and publish this!

**Comment by jywarren** [comment]
Er, sorry - is this ready to be merged? Given that it's causing an issue on the live site, I'd like to prioritize the narrower fix now, and if you'd like to expand on it in a follow-up, that's great! 

Thanks for responding so quickly to this!

> Also, it looks like all layers are not included in the map page here: https://publiclab.org/map

I'm not sure, but that page is still new - i think it should have the new menu and all available layers! 

Thank you!

**Comment by crisner** [comment]
Well, as a narrower fix it is ready to merge. This may probably need a follow up when I implement the new menu. Thanks!

**Comment by jywarren** [comment]
Thank you!!!

**Comment by crisner** [comment]
@jywarren, I see this has been published! :tada:
I bumped LEL in my forked copy of the plots2 repo to check if the changes have been reflected correctly and it indeed has! Awesome! :tada: 
- checked embed code in /map
- checked if layers are displayed by default
- checked for errors from LEL in the console

**Comment by crisner** [comment]
Also, I checked the issue with the fontawesome icon not displaying on the map. Installing this package: https://www.npmjs.com/package/@fortawesome/fontawesome-free, and adding the stylesheet to the `application.html.erb` does the trick. If you are okay with this I could perhaps open an FTO issue on plots2.

File: src/AllLayers.js [hunk 1]
```diff
@@ -133,6 +137,14 @@ L.LayerGroup.environmentalLayers = L.LayerGroup.extend(
         // Update map state from hash
         hash.update(this.options.currentHash);
       }
+        
+      for (let layer of this.options.layers.include) {
+        if (!this.options.addLayersToMap) {
```
Comment (sagarpreet-chadha) [inline src/AllLayers.js]: How did this logic got removed?

Comment (crisner) [reply src/AllLayers.js]: These lines were added. The ones that were removed are here: https://github.com/publiclab/leaflet-environmental-layers/blame/00c10f2c157a90a1d98a0254a061f9d6dd8f4c57/src/AllLayers.js#L123-L125

**Comment by jywarren** [comment]
adding the fontawesome would be great. Thanks!

**Pull request merged**